drivers/flash/soc_flash_mcux.c

```diff
@@ -293,7 +296,7 @@ static int flash_mcux_init(const struct device *dev)
 
 	rc = FLASH_Init(&priv->config);
 
-#ifdef SOC_HAS_IAP
+#if defined(SOC_HAS_IAP) || defined(CONFIG_SOC_FAMILY_NXP_MCX)
```

Comment:
Could we instead check this like `DT_NODE_HAS_STATUS(DT_INST(0, nxp_iap_mcx), okay)`? Also, do we need the family binding in this flash controller name?

Reply:
Update the binding name to be more generic.